src/OPL3.cpp

```diff
@@ -139,17 +139,32 @@ byte OPL3::getChipRegisterOffset(short reg) {
 }
 
 
+/**
+ * Write a given value to a chip wide register.
+ *
+ * @param synthUnit - The chip to address [0, 1]
+ * @param reg - The 9-bit register to write to.
+ * @param value - The value to write to the register.
+ */
+void OPL3::setChipRegister(short baseRegister, byte value) {
+	chipRegisters[getChipRegisterOffset(baseRegister)] = value;
+
+	byte bank = (baseRegister >> 8) & 0x01;
+	write(bank, baseRegister & 0xFF, value);
+}
+
+
 /**
  * Write a given value to a channel based register.
  *
  * @param baseRegister - The base address of the register.
  * @param channel - The channel to address [0, 17]
  * @param value - The value to write to the register.
  */
-void OPL3::setChannelRegister(byte baseRegister, byte channel, byte value) {
+void OPL3::setChannelRegister(short baseRegister, byte channel, byte value) {
```

Comment:
Using a `short` here could give the impression that the `baseRegister` is also used to select between bank 0 or 1, while this is determined by the `channel`. The `baseRegister` for a channel can only be 0xA0, 0xB0 or 0xC0.

Reply:
same here, I borrowed it from the other class. Fixed in new force-pushed commit